### ISSUE TITLE
Tab Page - from CustomerOrders example

### ISSUE THREAD

**Issue by stickybro** [volkanceylan/Serenity#41]
Hi Volkan,

I have a problem with a tab page like the example in serenity template - Customers -> Customer Orders. I have referred back to the example and it has the same problem :)

When you add a new Customer the Orders tab page is enabled but the CustomerOrdersGrid shows everyones orders instead of no orders. I have tried overriding the OnSaveSuccess function and apply the response.EntityID to the CustomerFilter but it doesn't fix it.

Can you help please?

**Comment by volkanceylan** [comment]
Oh my, i must have missed that. It also loads all order details when you just click the New button (watch ajax).

It's because Customer filter in opening grid (CustomerGrid) has a Select2 editor and it has empty string value instead of null when its actually empty.


But i checked for null, thats the problem in CustomerOrdersGrid:

```cs
        protected override bool GetGridCanLoad()
        {
            return base.GetGridCanLoad() && CustomerID != null;
        }
```

Just replace with:

```cs
        protected override bool GetGridCanLoad()
        {
            return base.GetGridCanLoad() && !string.IsNullOrEmpty(CustomerID);
        }
```

But problem still not solved, because Order dialog has a customer dropdown but that dropdown doesn't have that new customer, so you can't add a new order. Need to refresh its lookup.

I made several changes in CustomerOrdersGrid and CustomerDialog to fix this. Please check latest Serene commit, till i release a new version soon.

**Comment by stickybro** [comment]
Thank you :) loving this!!

**Comment by volkanceylan** [comment]
Fixed in Serene 1.6.3.4.

You are welcome ;)
